### PR TITLE
99 - updated build steps

### PR DESCRIPTION
**Pull request by micmro**
This still needs some final testing but I want to do this via `master` and wanted to get your feedback first.

- cleaned up grunt task-configs into separate files
- separated build and staging folders in `./build/` folder (not in git)
- release made via `release` branch
- sermver release helpers added

I think once this is tested to change version to `1.0.0` and use regular semver updates afterwards.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -1,219 +1,67 @@
...
 
-  require("load-grunt-tasks")(grunt);
 
   /** Version banner for static files (keep version format for "grunt-bump") */
-  var banner = "/*! github.com/micmro/PerfCascade Version:<%= pkg.version %> <%= grunt.template.today(\"(dd/mm/yyyy)\") %> */\n";
+  const banner = "/*! github.com/micmro/PerfCascade Version:<%= package.version %> <%= grunt.template.today(\"(dd/mm/yyyy)\") %> */\n";
+  let releaseIncrement = ["major", "minor"]
+    .indexOf(grunt.option('release-increment')) != -1 ? grunt.option('release-increment') : "patch";
 
-  grunt.initConfig({
```
Comment (micmro) [inline Gruntfile.js:9]: moved all configurations to files in `grunt-config/`

**Comment by tobli** [comment]
Looks awesome! I'll have a closer look and test it a bit soon.

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
```
Comment (micmro) [inline release.sh:1]: Could not find a grunt task that did exactly what we want (in a simple way - either without branch or release) do I just scripted it for now - we might replace this later.

File: src/ts/main.ts [hunk 3]
```diff
@@ -72,4 +72,4 @@ export { fromPerfCascadeFormat }
 export { transformHarToPerfCascade }
 export { makeLegend }
 // export typings
-export * from "./typing/index.d"
+export * from "./typing/index"
```
Comment (micmro) [inline src/ts/main.ts:75]: all `*.ts` and `*.d.ts` changes are to make it interfaces and typings exportable via `run:tscEs6`

File: package.json [hunk 4]
```diff
@@ -12,6 +12,8 @@
     "clearJS": "find ./src/ts -iname \"*.js\" -delete && find ./src/ts -iname \"*.map\" -delete",
     "ghPages": "grunt ghPages",
     "release": "grunt release",
```
Comment (micmro) [inline package.json:14]: default releases as patch `0.0.1` -> `0.0.2`

File: package.json [hunk 5]
```diff
@@ -12,6 +12,8 @@
...
+    "release-minor": "grunt release --release-increment=minor",
```
Comment (micmro) [inline package.json:15]: releases as minor `0.0.1` -> `0.1.0`

File: package.json [hunk 6]
```diff
@@ -12,6 +12,8 @@
...
+    "release-major": "grunt release --release-increment=major",
```
Comment (micmro) [inline package.json:16]: releases as major `0.0.1` -> `1.0.0`

**Comment by micmro** [comment]
@tobli @soulgalore, I have some time over the lunar new year now so I'm going to merge this to test it on `master` (releasing the tests as `v0.3.x` for now), it would be great if you can still review this PR anyway.
If all works and you're happy we could release a `v1.0.0` afterwards.

**Comment by micmro** [comment]
Fixes #99 and #116.

Seems to all work fine.

**Pull request merged**